[23기 최정민] ADD:로그인, 회원가입 모던창 , 로그인된 모던창 레이아웃 및 상세페이지 기능 구현  ,코드리뷰 반영

minmin9324 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
-로그인 회원가입 레이아웃 및 기능 완료 , 세부사항 페이지 data불러와서 UI그리기 ,리뷰댓글기능
		 
## 수정 사항들 자세한 내용
-기능구현
: 로그인,회원가입  기능구현 및 테스트 완료
: 새 리뷰 DB에 저장 구현 및 테스트 완료
: 상세페이지- 가게 상세정보 및 리뷰 data 불러와서 UI에 그리도록  로직 수정  및  테스트완료
-레이아웃 : 회원가입,로그인,로그인된 상태 모던창 완료 
(내정보버튼 ->로그인버튼->내정보버튼-> 로그인된 상태 모던 창
내정보버튼 ->회원가입버튼->회원가입하기버튼->내정보버튼-> 로그인된 상태 모던 창
내정보버튼 ->회원가입버튼->모더창 외 배경 클릭-> 로그인 모던 창->모더창 외 배경 클릭->모던창 꺼짐
내정보버튼 ->모더창 외 배경 클릭 -> 모던 창 꺼짐
내정보버튼 -> 로그아웃 버튼 -> 로그인 모던창)
-코드리뷰 수정 일부 완료(오타수정,리뷰 별점계산 구조 수정  ,카멜케이스 수정
,import 속성 순서 수정,삼항 연산자 수정 )
-config 파일 생성 , nav바 수정사항 (모달창 , a태그 button으로 변경 및 scss 수정)
## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
ADD: 식당 상세페이지 , 리뷰작성페이지 레이아웃 및 평점버튼기능

Commit:
MODIFY:이미지 수정

Tonyk0901 commented on src/Routes.js:
```diff
@@ -14,6 +15,11 @@ class Routes extends React.Component {
           <Route exact path="/FilterList" component={FilterList} />
           <Route exact path="/ShopList" component={ShopList} />
           <Route exact path="/ShopDetail" component={ShopDetail} />
+          <Route
+            exact
+            path="/ShopDetail-reveiwWritingPage"
+            component={ReveiwWritingPage}
+          />
```
- 오타있어요! `ShopDetail-reveiwWritingPage` -> `ShopDetail-reviewWritingPage`
- 일반적으로 경로이름은 소문자로 작성합니다! 띄어쓰기 부분은 `-` 로 연결해주세요!
- 예를 들어,
- `FilterList` -> `filterlist`

minmin9324 replied:
수정했습니다 !

Tonyk0901 commented on src/Routes.js:
```diff
@@ -4,6 +4,7 @@ import Main from './pages/Main/Main';
 import FilterList from './pages/FilterList/FilterList';
 import ShopList from './pages/ShopList/ShopList';
 import ShopDetail from './pages/ShopDetail/ShopDetail';
+import ReveiwWritingPage from './pages/ShopDetail/ReveiwWritingPage/ReveiwWritingPage';
```
- review 오타있어요!

Tonyk0901 commented on src/pages/ShopDetail/RestaurantInfo/RestaurantInfo.js:
```diff
@@ -0,0 +1,51 @@
+class RestaurantInfo extends React.Component {
+  render() {
+    return (
+      <table className="restaurantInfo">
```
- 👍 👍

Tonyk0901 commented on src/pages/ShopDetail/RestaurantInfo/RestaurantInfo.scss:
```diff
@@ -0,0 +1,24 @@
+    color: rgba(79, 79, 79, 0.6);
+    line-height: 1.7;
+    text-align: left;
+    padding-top: 5px;
+    padding-right: 10px;
+    padding-bottom: 5px;
```
- 이렇게 여러줄에 padding 속성을 부여하실 때는 한줄 속성 이용해주세요!

Tonyk0901 commented on src/pages/ShopDetail/RestaurantReview/RestaurantReview.js:
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import './RestaurantReview.scss';
+import ReveiwGradeButton from '../ReveiwWritingPage/ReveiwGradeButton/ReveiwGradeButton';
```
- import 속성 순서 신경써주세요!
- `React → Library(Package) → Component → 변수 / 이미지 → css 파일(scss 파일)`
- css 는 가장 마지막 입니다!
```suggestion
import React from 'react';
import ReveiwGradeButton from '../ReveiwWritingPage/ReveiwGradeButton/ReveiwGradeButton';
import './RestaurantReview.scss';
```

Tonyk0901 commented on src/pages/ShopDetail/RestaurantReview/RestaurantReview.scss:
```diff
@@ -0,0 +1,48 @@
+  }
+  .RestaurantReviewText {
+    margin-right: 30px;
+  }
```
- 네스팅 잘 해주세요!
- 그리고 새로운 셀렉터를 시작하실 때는 항상 한칸 띄워주세요!

Tonyk0901 commented on src/pages/ShopDetail/RestaurantReview/RestaurantReview.js:
```diff
@@ -0,0 +1,34 @@
+          </div>
+          <p className="RestaurantReviewText">너무 맛있어 !</p>
+        </div>
+      </li>
```
- className 주실 때는 첫글자를 소문자로 작성해주세요! 첫글자를 대문자로 작성하시는 경우는 컴포넌트 이름을 지으실 때 뿐 입니다!
- 추가적으로 항상 RestaurantReview 를 접두사로 붙여서 클래스네임을 부여하고 계신데, 명확하게 구분하기 위함은 알겠지만 불필요하게 클래스네임이 늘어나는 것 같습니다! 굳이 안 붙여주셔도 될 것 같아요!

Tonyk0901 commented on src/pages/ShopDetail/RestaurantReview/RestaurantReview.scss:
```diff
@@ -0,0 +1,48 @@
+.RestaurantReview {
+  display: flex;
+  padding-top: 50px;
+  padding-bottom: 50px;
```
- 한줄 속성 사용해주세요!

Tonyk0901 commented on src/pages/ShopDetail/ReveiwWritingPage/ReveiwWritingPage.js:
```diff
@@ -0,0 +1,75 @@
+import ReveiwGradeButton from './ReveiwGradeButton/ReveiwGradeButton';
+import './ReveiwWritingPage.scss';
+import ReviewPostingButton from './ReveiwPostingButton/ReviewPostingButton';
+class ReveiwWritingPage extends React.Component {
```
```suggestion
import React from 'react';
import { withRouter } from 'react-router-dom';
import ReveiwGradeButton from './ReveiwGradeButton/ReveiwGradeButton';
import ReviewPostingButton from './ReveiwPostingButton/ReviewPostingButton';
import './ReveiwWritingPage.scss';

class ReveiwWritingPage extends React.Component {
```

Tonyk0901 commented on src/pages/ShopDetail/ReveiwWritingPage/ReveiwGradeButton/ReveiwGradeButton.js:
```diff
@@ -0,0 +1,41 @@
+class ReveiwGradeButton extends React.Component {
+  render() {
+    const { SelectGrade, gradeIconSrc } = this.props;
+    let GoodOrBad = '';
+    let classname = '';
+    switch (gradeIconSrc) {
+      case 51:
+        GoodOrBad = '맛있다';
+        classname = 'GradeButtonOnText';
+        break;
+      case 50:
+        GoodOrBad = '맛있다';
+        break;
+      case 31:
+        GoodOrBad = '괜찮다';
+        classname = 'GradeButtonOnText';
+        break;
+      case 30:
+        GoodOrBad = '괜찮다';
+        break;
+      case 11:
+        GoodOrBad = '별로';
+        classname = 'GradeButtonOnText';
+        break;
+      case 10:
+        GoodOrBad = '별로';
+        break;
+    }
```
- 여기서 GoodOrBad 와 className 을 계산해주시는 것 보단, 상위 컴포넌트에서 전부 구성을 해서 하위 컴포넌트로 내려주는 구조를 취해주시는 것이 더 좋아보입니다!
- 예로 다음과 같이 구성될 수 있을 것 같습니다!
```jsx

  render() {
    const { SelectGrade, gradeIconSrc, content, isButtonClicked } = this.props;

    return (
      <button className="RestaurantReveiwGradeButton" onClick={SelectGrade}>
        <img alt="gradeIcon" src={`/images/shopDetail/${gradeIconSrc}.svg`} />
        <span className={isButtonClicked ? 'GradeButtonOnText' : ''}>
          {content}
        </span>
      </button>
    );
  }
```

minmin9324 replied:
오옷 !!

Tonyk0901 commented on src/pages/ShopDetail/ReveiwWritingPage/ReveiwWritingPage.js:
```diff
@@ -0,0 +1,75 @@
+          </div>
+          <div className="RestaurantReveiwBox">
+            <div className="RestaurantReveiwGrade">
+              <ReveiwGradeButton
+                SelectGrade={() => this.SelectGrade(5)}
+                gradeIconSrc={grade === 5 ? 51 : 50}
+              />
```
- 위에 드린 리뷰와 같이 구성을 한다면, 다음과 같이 작성될 것 같습니다.
```jsx
<ReveiwGradeButton
  SelectGrade={() => this.SelectGrade(5)}
  gradeIconSrc={grade === 5 ? 51 : 50}
  content={'맛있다'}
  isButtonClicked={grade === 5}
/>
```
- 참고해 보시고, 나머지는 스스로 한번 완성시켜 보세요!

minmin9324 replied:
감사합니다 성훈님 ㅠ! 이런 방법이 있었군요

Tonyk0901 commented on src/pages/ShopDetail/ShopDetail.js:
```diff
@@ -1,10 +1,68 @@
+    this.props.history.push('/ShopDetail-reveiwWritingPage');
+  };
+
+  wanted = () => {
```
- state 의 wanted 라는 동일한 이름을 가진 값이 있기 때문에, 매우매우 좋지 않은 함수 이름 같습니다.
- 그리고 함수는 액션을 의미하기 때문에 동사형으로 네이밍 해주세요! ex) handleWanted 와 같이 말이죠!

Tonyk0901 commented on src/pages/ShopDetail/ShopDetail.js:
```diff
@@ -1,10 +1,68 @@
+                리뷰쓰기
+              </button>
+              <button
+                className={
+                  wanted ? ' reveiwAndLikeIcon wanted' : 'reveiwAndLikeIcon'
+                }
```
- 달라지는 부분에 대해서만 삼항 연산자를 적용해주세요!
```suggestion
                className={`reviewAndLikeIcon ${wanted ? 'wanted' : ''}`}
```

Tonyk0901 reviewed (commented):
고생하셨습니다 정민님-! 리뷰 확인해주세요!

- 브랜치 이름은 이제 부터는 `jungmin` 이 아니라, `feature/작업하시는_기능` 입니다! ex) `feature/productDetail`
- css 작성하실 때, 새로운 셀렉터를 시작하시는 경우 한칸 띄워주세요!
- ShopDetail 컴포넌트와 같이, 백엔드 데이터가 들어와서 UI 가 그려져야 하는 컴포넌트는 하드코딩으로 UI 를 만들어주시는 것이 아니라, 목업 데이터를 생성하시고 그것에 대해 map 함수를 사용해서 UI 를 만들어주셔야 이후에 백엔드와 합쳐보는 과정이 훨씬 더 매끄럽게 진행될 수 있습니다! 이 컴포넌트 목업 데이터를 만들어서 구성해주세요!
- 많은 부분에서 첫글자를 대문자로 시작하는 카멜 케이스 컨벤션을 사용하고 계신데, 첫글자를 대문자로 작성해 주시는 것은 컴포넌트 이름을 지을때만 사용해 주세요! 함수이름 혹은 html 요소에 클래스 네임을 주시는 것과 같은 경우는 소문자로 시작해 주세요!

Commit:
ADD:로그인, 회원가입 모던창 , 로그인된 모던창 레이아웃 및 기능 구현
-기능구현
: 로그인,회원가입  기능구현 및 테스트 완료
: 새 리뷰 DB에 저장 구현 및 테스트 완료
: 상세페이지- 가게 상세정보 및 리뷰 data 불러와서 UI에 그리도록  로직 수정  및  테스트완료
-레이아웃 : 회원가입,로그인,로그인된 상태 모던창 완료
-코드리뷰 수정 일부 완료(오타수정,리뷰 별점계산 구조 수정  ,카멜케이스 수정
,import 속성 순서 수정,삼항 연산자 수정 )

minmin9324 committed:
Merge branch 'main' into jungmin
